### ISSUE TITLE
Z-Wave-JS-UI Thermostats not Autodiscovering all devices - missing Cooling Setpoint, Fan Control, Fan State, Operating State.

### ISSUE THREAD

**Issue by eagle360** [domoticz/domoticz#6069]
Is there any way to manually add devices in Domoticz to control functions in MQTT that do not AutoDiscover?

Domoticz Version: 2024.4 zwave-js-ui: 9.12.0 zwave-js: 12.5.6

I have been working with two z-wave thermostats that both worked just fine with OZW in older Domoticz versions. Remotec ZTS-110 and CT100.  Moving to zwaves-js-ui neither of them work.  They are missing the Cooling Setpoint, Fan Control, Fan State, and Operating State devices in Domoticz (so basically useless).  I have tried all sorts or variations of the climate HASS and can not get any of these missing devices in Domoticz.  The HASS from zwave-js-ui is clearly missing the Fan Mode control but even if I add it and get it to update in MQTT it still does not generate a device in Domoticz. 

Domoticz 2024.4 with OZW also no longer works with these thermostats. I tried to include these same thermostats back into Domoticz using OZW (on a different Domoticz install) and they are no longer working in the new version of Domotiz.  They worked for years with Domtocz 2023.2 and prior with no issues but now Domoticz will not generate ANY Set points for them now.

I don't care about OZW anymore if I can just get the zwave-js-ui integration to work.  But it might be a related problem.

Also with zwave-js-ui the thermostat mode (Off, Heating, Cooling, Auto) is a Selector Switch device now instead of a Thermostat Mode device.  Switching this Selector switch does change the mode in the thermostat and the Heating set point and temperature from the thermostat also work.  The problem is just that the  Cooling Set point, Fan Control, Fan State, Operating State devices are all missing.

If I change the HASS "temperature_state_topic": "zwave/Test/ThermostatTest2/67/0/setpoint/1", to "temperature_state_topic": "zwave/Test/ThermostatTest2/67/0/setpoint/2", the Set point in Domoticz will then control the Cooling set point so I know that the HASS configuration is being read by Domoticz and can actually control the other set point.  It just needs to create a device that is mapped to it.

All the thermostat functions in zwave-js-ui are in the UI and can be controlled from there.  They are also all in the MQTT and I can see the values update in MQTT as well.  So the only issue is that Domoticz is not creating devices that map to these functions.

Is there any way to add devices manually to control MQTT devices that just don't AutoDiscover?

This is the climate HASS from zwave-js-ui:
```
{
  "type": "climate",
  "object_id": "climate",
  "discovery_payload": {
    "min_temp": 5,
    "max_temp": 40,
    "temp_step": 0.5,
    "modes": [
      "off",
      "heat",
      "cool",
      "auto"
    ],
    "mode_state_template": "{{ {0: \"off\", 1: \"heat\", 2: \"cool\", 3: \"auto\"}[value_json.value] | default('off') }}",
    "current_temperature_topic": "zwave/Test/ThermostatTest2/49/0/Air_temperature",
    "current_temperature_template": "{{ value_json.value }}",
    "temperature_state_topic": "zwave/Test/ThermostatTest2/67/0/setpoint/1",
    "temperature_state_template": "{{ value_json.value }}",
    "temperature_command_topic": "zwave/Test/ThermostatTest2/67/0/setpoint/1/set",
    "mode_state_topic": "zwave/Test/ThermostatTest2/64/0/mode",
    "mode_command_topic": "zwave/Test/ThermostatTest2/64/0/mode/set",
    "action_topic": "zwave/Test/ThermostatTest2/66/0/state",
    "action_template": "{{ {0: \"idle\", 1: \"heating\", 2: \"cooling\", 3: \"fan\", 4: \"idle\", 5: \"idle\", 6: \"fan\", 7: \"heating\", 8: \"heating\", 9: \"cooling\", 10: \"heating\", 11: \"heating\"}[value_json.value] | default('idle') }}",
    "temperature_unit": "F",
    "precision": 0.1,
    "device": {
      "identifiers": [
        "zwavejs2mqtt_0xe24caefc_node35"
      ],
      "manufacturer": "Remotec",
      "model": "ZTS-110 Z Wave Thermostat (ZTS-110)",
      "name": "Test-ThermostatTest2",
      "sw_version": "3.14"
    },
    "availability": [
      {
        "payload_available": "true",
        "payload_not_available": "false",
        "topic": "zwave/Test/ThermostatTest2/status",
        "value_template": "{{'true' if value_json.value else 'false'}}"
      },
      {
        "topic": "zwave/_CLIENTS/ZWAVE_GATEWAY-zwave-js-ui/status",
        "value_template": "{{'online' if value_json.value else 'offline'}}"
      },
      {
        "payload_available": "true",
        "payload_not_available": "false",
        "topic": "zwave/driver/status"
      }
    ],
    "availability_mode": "all",
    "name": "Test-ThermostatTest2_climate",
    "unique_id": "zwavejs2mqtt_0xe24caefc_Node35_climate"
  },
  "values": [
    "49-0-Air temperature",
    "64-0-mode",
    "67-0-setpoint-1",
    "67-0-setpoint-2",
    "67-0-setpoint-1",
    "66-0-state"
  ],
  "mode_map": {
    "off": 0,
    "heat": 1,
    "cool": 2,
    "auto": 3
  },
  "setpoint_topic": {
    "1": "67-0-setpoint-1",
    "2": "67-0-setpoint-2",
    "3": "67-0-setpoint-1"
  },
  "default_setpoint": "67-0-setpoint-1",
  "action_map": {
    "0": "idle",
    "1": "heating",
    "2": "cooling",
    "3": "fan",
    "4": "idle",
    "5": "idle",
    "6": "fan",
    "7": "heating",
    "8": "heating",
    "9": "cooling",
    "10": "heating",
    "11": "heating"
  },
  "discoveryTopic": "climate/Test-ThermostatTest2/climate/config",
  "persistent": false,
  "ignoreDiscovery": false,
  "id": "climate_climate"
}

```
Here are the MQTT topics:
```
zwave/Test/ThermostatTest2/49/0/Air_temperature
zwave/Test/ThermostatTest2/64/0/mode
zwave/Test/ThermostatTest2/64/0/manufacturerData
zwave/Test/ThermostatTest2/66/0/state
zwave/Test/ThermostatTest2/67/0/setpoint/1
zwave/Test/ThermostatTest2/67/0/setpoint/2
zwave/Test/ThermostatTest2/68/0/mode
zwave/Test/ThermostatTest2/69/0/state

domoticz/climate/Test-ThermostatTest2/climate/config

{"min_temp":5,"max_temp":40,"temp_step":0.5,"modes":["off","heat","cool","auto"],"mode_state_template":"{{ {0: \"off\", 1: \"heat\", 2: \"cool\", 3: \"auto\"}[value_json.value] | default('off') }}","current_temperature_topic":"zwave/Test/ThermostatTest2/49/0/Air_temperature","current_temperature_template":"{{ value_json.value }}","temperature_state_topic":"zwave/Test/ThermostatTest2/67/0/setpoint/1","temperature_state_template":"{{ value_json.value }}","temperature_command_topic":"zwave/Test/ThermostatTest2/67/0/setpoint/1/set","mode_state_topic":"zwave/Test/ThermostatTest2/64/0/mode","mode_command_topic":"zwave/Test/ThermostatTest2/64/0/mode/set","action_topic":"zwave/Test/ThermostatTest2/66/0/state","action_template":"{{ {0: \"idle\", 1: \"heating\", 2: \"cooling\", 3: \"fan\", 4: \"idle\", 5: \"idle\", 6: \"fan\", 7: \"heating\", 8: \"heating\", 9: \"cooling\", 10: \"heating\", 11: \"heating\"}[value_json.value] | default('idle') }}","temperature_unit":"F","precision":0.1,"device":{"identifiers":["zwavejs2mqtt_0xe24caefc_node35"],"manufacturer":"Remotec","model":"ZTS-110 Z Wave Thermostat (ZTS-110)","name":"Test-ThermostatTest2","sw_version":"3.14"},"availability":[{"payload_available":"true","payload_not_available":"false","topic":"zwave/Test/ThermostatTest2/status","value_template":"{{'true' if value_json.value else 'false'}}"},{"topic":"zwave/_CLIENTS/ZWAVE_GATEWAY-zwave-js-ui/status","value_template":"{{'online' if value_json.value else 'offline'}}"},{"payload_available":"true","payload_not_available":"false","topic":"zwave/driver/status"}],"availability_mode":"all","name":"Test-ThermostatTest2_climate","unique_id":"zwavejs2mqtt_0xe24caefc_Node35_climate"}
```

Let me know if you need anything else.  Again if I could just add these devices manually I would not be dead in the water like I am now.   Thanks.

**Comment by gizmocuz** [comment]
What is HASS? (HASS configuration)

Could you please follow this guide how to report the needed topics/payloads

https://www.domoticz.com/wiki/MQTT_AD_Report_Problems

**Comment by eagle360** [comment]
Sorry I thought I did.  What did I not include?

Here are the payloads I didn't include above:
```
zwave/Test/ThermostatTest2/49/0/Air_temperature
{"time":1714737272882,"value":67.5}
zwave/Test/ThermostatTest2/64/0/mode
{"time":1714736660966,"value":3}
zwave/Test/ThermostatTest2/64/0/manufacturerData
{"time":1714736309188}
zwave/Test/ThermostatTest2/66/0/state
{"time":1714736676875,"value":1}
zwave/Test/ThermostatTest2/67/0/setpoint/1
{"time":1714736691068,"value":72}
zwave/Test/ThermostatTest2/67/0/setpoint/2
{"time":1714739306372,"value":80}
zwave/Test/ThermostatTest2/68/0/mode
{"time":1714736622411,"value":0}
zwave/Test/ThermostatTest2/69/0/state
{"time":1714614318217,"value":0}
```

**Comment by eagle360** [comment]
Also I don't know what HASS is it looks like (Home Assistant Server Service) . In z-wave-js-ui it says "Hass device JSON" and is what is used for the AutoDiscover config topic.

**Comment by gizmocuz** [comment]
Thanks for the extra payloads. Hmm did not know about HASS... Should be called ZWaveJS-UI config maybe... 😁

**Comment by gizmocuz** [comment]
Okey, so you can set the thermostat mode, and also see the actual temperature and control the temperature.
Seems like you can already do a lot...

Not working:
- Cooling Set point (I don't see a cooling setpoint defined .. no setpoint/2)
Is there another config object that creates a thermostat device? (for setpoint/2)
If not, you could ask ZWaveJS-UI to let this create for this device, or you can add one to MQTT yourself manually

Below an example config topic for a setpoint device

```
{
  "temp_stat_t": "~/ess_setpoint/state",
  "dev": {
    "manufacturer": "PA1DVB",
    "ids": [
      "0x8000000002"
    ]
  },
  "~": "my_victron",
  "name": "ESS Setpoint",
  "uniq_id": "0x8000000002",
  "temp_unit": "Watt",
  "temp_cmd_t": "~/set/ess_setpoint",
  "temp_stat_tpl": "{{ value_json.ess_setpoint }}",
  "temp_step": 100,
  "min_temp": -5000,
  "max_temp": 5000
}
```

- Fan Control
If this is other then the 'modes', these are also missing.
You could create a 'select' MQTT-AD config like

```
{
  "stat_t": "~/ess_mode/state",
  "dev": {
    "manufacturer": "PA1DVB",
    "ids": [
      "0x8000000001"
    ]
  },
  "~": "my_victron",
  "name": "ESS Control Mode",
  "options": [
    "Mode 1",
    "Mode 2"
  ],
  "cmd_t": "~/set/ess_mode",
  "uniq_id": "0x8000000001",
  "val_tpl": "{{ value_json.ess_mode }}"
}
```
- Fan State (will be solved when using your own like above)

- Operating State (I assume this is the action_topic/action_template ?)

This is a very complicated devices as it has lots of functionality.
Actually if ZWaveJS-UI just splits it up into (config) pieces, it would be possible

**Comment by eagle360** [comment]
Thanks. I don't completely understand the AutoDiscover configurations or how to manually add new ones. Can't find any useful documentation on the climate device.

Can what you are suggesting be done with the MQTT explorer?  I think I have seen CLI to add topics to MQTT but either way I would need to know what to name the topic.

The current config topic is:
domoticz/climate/Test-ThermostatTest2/climate/config
Not sure what to call new ones since they can't have the same name and this all seems to be under climate (haven't been able to figure out all the naming conventions).

So far all I have tried is changing the HASS device JSON in z-wave-js-ui.  And I don't see any way to add anything there expect to change an existing HASS device JSON.   I was able to fix a different issue with my Z-wave Fan Switch by changing the "speed_command_topic": and "speed_state_topic": that Domoticz didn't like to "percentage_command_topic": and "percentage_state_topic":.  That worked and then Domoticz added the device and it has been working.

But for the Thermostat I tried to add the Fan Modes and the cooling set point in the HASS device JSON but nothing has worked. For the Thermostat Fan control I found some examples and I came up with and tried the following (but it did not work); 

```
{
  "type": "climate",
  "object_id": "climate",
  "discovery_payload": {
    "min_temp": 5,
    "max_temp": 40,
    "temp_step": 0.5,
    "modes": [
      "off",
      "heat",
      "cool",
      "auto"
    ],
    "fan_modes": [
      "Auto low",
      "Low"
    ],
    "mode_state_template": "{{ {0: \"off\", 1: \"heat\", 2: \"cool\", 3: \"auto\"}[value_json.value] | default('off') }}",
    "fan_mode_state_template": "{{ {0: \"Auto Low\", 1: \"Low\"}[value_json.value] | default('Auto Low') }}",
    "current_temperature_topic": "zwave/Test/ThermostatTest2/49/0/Air_temperature",
    "current_temperature_template": "{{ value_json.value }}",
    "temperature_state_topic": "zwave/Test/ThermostatTest2/67/0/setpoint/1",
    "temperature_state_template": "{{ value_json.value }}",
    "temperature_command_topic": "zwave/Test/ThermostatTest2/67/0/setpoint/1/set",
    "fan_mode_state_topic": "zwave/Test/ThermostatTest2/68/0/mode",
    "fan_mode_command_topic": "zwave/Test/ThermostatTest2/68/0/mode/set",
    "mode_state_topic": "zwave/Test/ThermostatTest2/64/0/mode",
    "mode_command_topic": "zwave/Test/ThermostatTest2/64/0/mode/set",
    "action_topic": "zwave/Test/ThermostatTest2/66/0/state",
    "action_template": "{{ {0: \"idle\", 1: \"heating\", 2: \"cooling\", 3: \"fan\", 4: \"idle\", 5: \"idle\", 6: \"fan\", 7: \"heating\", 8: \"heating\", 9: \"cooling\", 10: \"heating\", 11: \"heating\"}[value_json.value] | default('idle') }}",
    "temperature_unit": "F",
    "precision": 0.1,
    "device": {
      "identifiers": [
        "zwavejs2mqtt_0xe24caefc_node32"
      ],
      "manufacturer": "Remotec",
      "model": "ZTS-110 Z Wave Thermostat (ZTS-110)",
      "name": "Test-ThermostatTest2",
      "sw_version": "3.14"
    },
    "name": "Test-ThermostatTest2_climate",
    "unique_id": "zwavejs2mqtt_0xe24caefc_Node32_climate"
  },
  "values": [
    "49-0-Air temperature",
    "64-0-mode",
    "67-0-setpoint-1",
    "67-0-setpoint-2",
    "67-0-setpoint-1",
    "69-0-state",
    "68-0-mode"
  ],
  "mode_map": {
    "off": 0,
    "heat": 1,
    "cool": 2,
    "auto": 3
  },
  "fan_mode_map": {
    "Auto low": 0,
    "Low": 1
  },
  "setpoint_topic": {
    "1": "67-0-setpoint-1",
    "2": "67-0-setpoint-2",
    "3": "67-0-setpoint-1"
  },
  "default_setpoint": "67-0-setpoint-1",
  "action_map": {
    "0": "idle",
    "1": "heating",
    "2": "cooling",
    "3": "fan",
    "4": "idle",
    "5": "idle",
    "6": "fan",
    "7": "heating",
    "8": "heating",
    "9": "cooling",
    "10": "heating",
    "11": "heating"
  },
  "discoveryTopic": "climate/Test-ThermostatTest2/climate/config",
  "persistent": true,
  "ignoreDiscovery": false,
  "id": "climate_climate"
}
```

Is the only way to manually add devices to Domoticz, that can control MQTT topics like the Cooling set point and Fan, by adding a config topic to MQTT or is there some way to do it directly in Domoticz and completely by-pass the AutoDiscover?

Thanks.

**Comment by gizmocuz** [comment]
A good starting point is here:

https://www.home-assistant.io/integrations/mqtt/

Yes, you can do this with MQTT Explorer, just make sure that the 'config' objects are created with the 'persistent' flag.
This means when MQTT is restarted or when anyone reconnect to the MQTT broker it receives the config again

For now, yes, I think the only way is to add some manual configs until ZWaveJS does this for you.

It is not that hard

**Comment by eagle360** [comment]
**It appears that Domoticz still is not doing what it should be as it won't add a Fan Mode device with the configuration above.**

Also I understand it might not be that hard but the documentation is lacking.  As I said I can't find anything on the "climate" configurations at all.  I don't see anything for set points or modes.

Just even knowing what to name the topic is not clear.

The current topic is `domoticz/climate/Test-ThermostatTest2/climate/config`
But if I want to make a cooling topic for the cooling set point what would it be?
`<discovery_prefix>/<component>/[<node_id>/]<object_id>/config`
`<component>` seems to be a special name such as `binary_sensor` in this case `climate` Ok so `domoticz/climate/` `[<node_id>/]` doesn't seam to matter so  `domoticz/climate/Test-ThermostatTest2/` looks like `<object_id>` might be the one to change.  Is the format of this value important???  We can just try `climate2` so that would get us
???`domoticz/climate/Test-ThermostatTest2/climate2/config`???

I tried creating this new topic from MQTT
`domoticz/climate/Test-ThermostatTest2/climate2/config` 

Ok after some messing around I finally figured out that the new config topic was actually changing the configuration of the existing heating setpoint device to the cooling setpoint topic. So then after some trial and error I discover that changing
`"unique_id": "zwavejs2mqtt_0xe24caefc_Node35_climate"`
to
`"unique_id": "zwavejs2mqtt_0xe24caefc_Node35_climate2"`
is what will make Domoticz generate a unique device (vs just changing the configuration of the existing heating set point device).  This finally got Domoticz to create a new device that is mapped to the cooling setpoint and after rediscovering the climate topic from Z-wave-JS-UI the other setpoint maps back to the heating setpoint so now I have a Heating and Cooling setpoint.  **Progress! But still no Fan mode or Operating state.**

Config topic that creates a new Cooling setpoint but still no Fan Mode device or Operating state.
```
{
  "min_temp": 41,
  "max_temp": 104,
  "temp_step": 1,
  "fan_modes": [
    "Auto low",
    "Low"
  ],
  "fan_mode_state_template": "{{ {0: \"Auto Low\", 1: \"Low\"}[value_json.value] | default('Auto Low') }}",
  "temperature_state_topic": "zwave/Test/ThermostatTest2/67/0/setpoint/2",
  "temperature_state_template": "{{ value_json.value }}",
  "temperature_command_topic": "zwave/Test/ThermostatTest2/67/0/setpoint/2/set",
  "fan_mode_state_topic": "zwave/Test/ThermostatTest2/68/0/mode",
  "fan_mode_command_topic": "zwave/Test/ThermostatTest2/68/0/mode/set",
  "action_topic": "zwave/Test/ThermostatTest2/66/0/state",
  "temperature_unit": "F",
  "precision": 0.1,
  "device": {
    "identifiers": [
      "zwavejs2mqtt_0xe24caefc_node35"
    ],
    "manufacturer": "Remotec",
    "model": "ZTS-110 Z Wave Thermostat (ZTS-110)",
    "name": "Test-ThermostatTest2",
    "sw_version": "3.14"
  },
  "name": "Test-ThermostatTest2_climate2",
  "unique_id": "zwavejs2mqtt_0xe24caefc_Node35_climate2"
}
```

I then thought maybe the `modes` is generic and can be assigned to the fan mode topic instead so I tried:

```
{
  "min_temp": 41,
  "max_temp": 104,
  "temp_step": 1,
  "modes": [
    "Auto low",
    "Low"
  ],
  "mode_state_template": "{{ {0: \"Auto Low\", 1: \"Low\"}[value_json.value] | default('Auto Low') }}",
  "temperature_state_topic": "zwave/Test/ThermostatTest2/67/0/setpoint/2",
  "temperature_state_template": "{{ value_json.value }}",
  "temperature_command_topic": "zwave/Test/ThermostatTest2/67/0/setpoint/2/set",
  "mode_state_topic": "zwave/Test/ThermostatTest2/68/0/mode",
  "mode_command_topic": "zwave/Test/ThermostatTest2/68/0/mode/set",
  "action_topic": "zwave/Test/ThermostatTest2/66/0/state",
  "temperature_unit": "F",
  "precision": 0.1,
  "device": {
    "identifiers": [
      "zwavejs2mqtt_0xe24caefc_node35"
    ],
    "manufacturer": "Remotec",
    "model": "ZTS-110 Z Wave Thermostat (ZTS-110)",
    "name": "Test-ThermostatTest2",
    "sw_version": "3.14"
  },
  "name": "Test-ThermostatTest2_climate2",
  "unique_id": "zwavejs2mqtt_0xe24caefc_Node35_climate2"
}
```
This generated this Domoticz error:
`2024-05-04 12:16:44.338 Error: MQTT Broker: Climate device invalid/unknown mode received! (zwavejs2mqtt_0xe24caefc_Node35_climate2: Auto Low)`

So clearly Domoticz has issues with Thermostat `climate` fan modes.  And still no Operating state from the topic `zwave/Test/ThermostatTest2/66/0/state`

Any ideas for the Fan Modes and Operating state?
Thanks.

**Comment by gizmocuz** [comment]
All MQTT-AD parameters are documented on the page I mentioned above.
Domoticz has no issues

I would suggest todo some debugging/testing with MQTT Explorer, add a second MQTT-AD hardware in domoticz with another discovery topic so you can experiment
Or install a second domoticz instance if you are using docker

**Comment by eagle360** [comment]
I found that documentation page myself, initially looking for a solution, and is it mostly useless in this case.  Again nothing related to `climate` topics `Setpoints`, `modes`, or `Operating Status` .  Part of the question is how Domoticz implements the autodiscovery. Is there any documentation for that?

I assume there is a way to include both heating and cooling setpoints in a single HASS Device JSON config but since Domoticz does not seem to use the name in the config `"temperature_state_topic":` to create an unique setpoint there can only be one setpoint per config.
`"temperature_state_topic": "zwave/Test/ThermostatTest2/67/0/setpoint/1",`
There is no way to have a second setpoint 
`"temperature2_state_topic": "zwave/Test/ThermostatTest2/67/0/setpoint/2",`

The mode has the same issue since `"mode_state_topic":` is not unique in Domoticz so no way to add a second mode
`"fan_mode_state_topic":` Doesn't work.

Also clearly Domoticz can't accept a Fan mode config as it generates:
`Error: MQTT Broker: Climate device invalid/unknown mode received! (zwavejs2mqtt_0xe24caefc_Node35_climate2: Auto Low)`

I did finally just use the following config to hack a Fan mode device.  The interesting thing was that even though Domoticz give the error above it actually did create a device with `Auto Low` and `Low` but just never made it available.   I know this because after using the config below it showed the new Fan mode device with `Auto Low` and `Low` until I clicked on it and then it changed instantly to the values of `Off` and `Low` (per the new config).

```
{
  "min_temp": 41,
  "max_temp": 104,
  "temp_step": 1,
  "modes": [
    "Off",
    "Low"
  ],
  "mode_state_template": "{{ {0: \"Off\", 1: \"Low\"}[value_json.value] | default('Off') }}",
  "temperature_state_topic": "zwave/Test/ThermostatTest2/67/0/setpoint/2",
  "temperature_state_template": "{{ value_json.value }}",
  "temperature_command_topic": "zwave/Test/ThermostatTest2/67/0/setpoint/2/set",
  "mode_state_topic": "zwave/Test/ThermostatTest2/68/0/mode",
  "mode_command_topic": "zwave/Test/ThermostatTest2/68/0/mode/set",
  "temperature_unit": "F",
  "device": {
    "identifiers": [
      "zwavejs2mqtt_0xe24caefc_node35"
    ],
    "manufacturer": "Remotec",
    "model": "ZTS-110 Z Wave Thermostat (ZTS-110)",
    "name": "Test-ThermostatTest2",
    "sw_version": "3.14"
  },
  "name": "Test-ThermostatTest2_climate2",
  "unique_id": "zwavejs2mqtt_0xe24caefc_Node35_climate2"
}
```

Also `climate` `modes` are still generating `Selector switch` devices and not `climate mode` devices.  So yes Domoticz still has issues.

And there still is no Operating State device generated by Domoticz.

If "Domoticz has no issues" then can you **please** just share the config JSON that will actually get Domoticz to add these devices.

The only one I am missing now is the Operating State.  If you don't know how to do this I can see if I can get a Z-wave sensor of some type that has discrete steps that it reports and see what the HASS Device JSON for that looks like.  Maybe it could be read as just a value as well.

I'm not sure why issues would not be addressed as issues vs just saying there are no issues when there clearly are. If they are  working then an actual working example would be good to show that yes indeed they are working.

**Comment by eagle360** [comment]
Well after much searching I finally came across this link https://www.home-assistant.io/integrations/climate.mqtt/ and it clearly confirms that the MQTT-AD in Domoticz is just not working correctly.

Also that link is only linked via the page listed above in a very convoluted manor.  I only figured it out because I finally found the actual page and worked backwards looking through the page source of the other link.  You have to click on Climate and then you get a page of a bunch of brand images and you then need to find MQTT-HVAC in amongst the hundreds of options.  In the future the direct link would be more helpful.

**Comment by gizmocuz** [comment]
You can't blame me/us for the fact that you cannot find the correct documentation or that you find it hard to navigate in the home assistant documentation

Maybe you should have a look in our code and make some modifications to it if you think it is incorrect.
Regarding extra modes, I already mentioned how to add these, these can be done via a 'select' type, or a second climate device
There is no 'temperature2_state_topic', and you are correct, we also do not support this (as do other implementers)
This device is also a complicated device as it contains a lot of features.
If they were split in different configs live was more easy.

Anyway, I gave you some help/hints. This is not a one-stop-shopping channel. Also this is a opensource project so feel free to contribute with a PR that solves all your issues.

With my two configs above you can create thermostat setpoints and a mode switch

**Comment by eagle360** [comment]
I appreciate the help and you got me pointed in the right direction to come up with a work round.   I was just trying to give useful feedback.

I thought the point of this forum was to report issues so they can be known and potentially fixed in the future.  If nothing else it should help others to find a work around to fix their issues in the meantime.

The above config that I posted does add the cooling setpoint and a usable Fan mode switch.  I used the following `binary_sensor` config to get the Fan Status device.  I still just need the Operating state so I am still looking for an example of a mode or selector switch sensor (input only).  If I can't find one I will just use a sensor type that reads in a value and use that. 

`domoticz/binary_sensor/Test-ThermostatTest2/fan_status/config`
```
{
  "payload_on": 1,
  "payload_off": 0,
  "value_template": "{{ value_json.value }}",
  "state_topic": "zwave/Test/ThermostatTest2/69/0/state",
  "device": {
    "identifiers": [
      "zwavejs2mqtt_0xe24caefc_node35"
    ],
    "manufacturer": "Remotec",
    "model": "ZTS-110 Z Wave Thermostat (ZTS-110)",
    "name": "Test-ThermostatTest2",
    "sw_version": "3.14"
  },
  "name": "Test-ThermostatTest2_Fan_State",
  "unique_id": "zwavejs2mqtt_0xe24caefc_Node35_climate3"
}
```

Anyway I just wanted to report the current Domoticz MQTT-AD issues so maybe they can be addressed at some point or at least help other people to find a work around for them.
